server/src/bin/delete-worker.rs

```diff
@@ -216,13 +219,32 @@ async fn delete_worker(
         let delete_worker_message: DeleteMessage =
             serde_json::from_str(&serialized_message).expect("Failed to parse file message");
 
+        let dataset = get_dataset_by_id_query(
+            models::UnifiedId::TrieveUuid(delete_worker_message.dataset_id),
+            web_pool.clone(),
+        )
+        .await;
+
+        if dataset.is_err() {
+            let _ = readd_error_to_queue(
+                delete_worker_message,
+                dataset.err().unwrap(),
+                redis_pool.clone(),
+            )
+            .await;
+            return;
+        }
+
+        let server_config =
+            ServerDatasetConfiguration::from_json(dataset.unwrap().server_configuration);
```

Comment:
Instead of the `unwrap` in these we should use a `match` and handle the err correctly w/in the match so we can guarantee not crash the worker server processes.